cloud-sql-proxy/cloud-sql-proxy.sh

```diff
@@ -82,6 +82,53 @@ readonly ADDITIONAL_INSTANCES_KEY='attributes/additional-cloud-sql-instances'
 # Dataproc master nodes information
 readonly DATAPROC_MASTER=$(/usr/share/google/get_metadata_value attributes/dataproc-master)
 
+function get_java_property() {
+  local property_file=$1
+  local property_name=$2
+  local property_value=$(grep "^${property_name}=" "${property_file}" | \
+      tail -n 1 | cut -d '=' -f 2- | sed -r 's/\\([#!=:])/\1/g')
+  echo "${property_value}"
+}
+
+function get_dataproc_property() {
+  local property_name=$1
+  local property_value=$(get_java_property \
+    /etc/google-dataproc/dataproc.properties \
+    "${property_name}")
+  echo "${property_value}"
+}
+
+function is_component_selected() {
+  local component=$1
+
+  local activated_components=$(get_dataproc_property \
+      dataproc.components.activate)
+
+  if [[ ${activated_components} == *${component}* ]]; then
+    return 0
+  else
+    return 1
+  fi
+}
+
+readonly KERBEROS_ENABLED=$(is_component_selected 'kerberos' && echo 'true' || echo 'false')
+
+function get_hive_principal() {
+  local host=$(hostname)
+  local domain=$(dnsdomainname)
```

Comment:
Shouldn't you convert to all upper case?

Reply:
Done
Used fully qualified hostname, made realm upper case